tests/testthat/test_pipeop_filterrows.R

```diff
@@ -0,0 +1,90 @@
+
+  expect_datapreproc_pipeop_class(PipeOpFilterRows, task = task)
+})
+
+test_that("PipeOpFilterRows - NA handling", {
+  dat = iris
+  dat$Sepal.Length[c(1L, 3L, 100L)] = NA
+  dat$Petal.Width[c(1L, 4L)] = NA
+  dat$Petal.Length[5] = NA
+  dat$Species[2L] = NA
+  task = TaskClassif$new("test", backend = dat, target = "Species")
+
+  op = PipeOpFilterRows$new(param_vals = list(na_column = "Species"))
+
+  op$param_set$values$na_column = "Sepal.Length"
+  train_out1 = op$train(list(task))[[1L]]
+  expect_equal(op$state$na_ids, c(1, 3, 100))
+
+  op$param_set$values$na_column = "all"
+  train_out2 = op$train(list(task))[[1L]]
+  expect_equal(op$state$na_ids, c(1, 2, 3, 4, 5, 100))
+  predict_out2 = op$predict(list(task))[[1L]]
+  expect_equal(train_out2, predict_out2)
+
+  op$param_set$values$skip_during_predict = TRUE
+  expect_equal(op$predict(list(task))[[1L]], task)
+})
+
+test_that("PipeOpFilterRows - filter by column name", {
+  set.seed(1)
+  dat = iris
+  dat$filter = sample(c(FALSE, TRUE), size = 150, replace = TRUE)
+  task = TaskClassif$new("test", backend = dat, target = "Species")
+
+  op = PipeOpFilterRows$new(param_vals = list(filter = "filter"))
+
+  train_out1 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, which(dat$filter))
+  expect_equal(train_out1$data(), task$data(which(dat$filter)))
+
+  op$param_set$values$invert = TRUE
+  train_out2 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, which(!dat$filter))
+  expect_equal(train_out2$data(), task$data(which(!dat$filter)))
+})
+
+test_that("PipeOpFilterRows - filter by expression", {
+  task = mlr_tasks$get("iris")
+  task_cp = task$clone(deep = TRUE)
+
+  op = PipeOpFilterRows$new(param_vals = list(filter = expression(Sepal.Length < 6 & Petal.Width > 1)))
+  train_out1 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, which(iris$Sepal.Length < 6 & iris$Petal.Width > 1))
+
+  # zero indices left
+  op$param_set$values$filter = expression(Petal.Length < 3 & Petal.Width > 1)
+  train_out2 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, integer(0))
+  expect_equal(train_out2, task_cp$filter(integer(0)))
+  op$param_set$values$invert = TRUE
+  train_out3 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, 1:150)
+  expect_equal(train_out3, task)
+})
+
+test_that("PipeOpFilterRows - filter by ids", {
+  # in combination with NA
+  dat = iris
+  dat$Sepal.Length[c(1L, 3L, 100L)] = NA
+  dat$Petal.Width[c(1L, 4L)] = NA
+  dat$Petal.Length[5] = NA
+  dat$Species[2L] = NA
+  task = TaskClassif$new("test", backend = dat, target = "Species")
+
+  op = PipeOpFilterRows$new(param_vals = list(na_column = "all", filter = 1:10))
+  train_out1 = op$train(list(task))[[1L]]
+  expect_equal(op$state$row_ids, setdiff(1:10, c(1, 2, 3, 4, 5)))
+  op$param_set$values$invert = TRUE
+  train_out2 = op$train(list(task))[[1L]]
+  expect_equal(op$state$na_ids, c(1, 2, 3, 4, 5, 100))
+  expect_true(!(100 %in% op$state$row_ids))
+})
```

Comment:
We should test how `filterrows` works with learnercv. 
I think we'd like a way to only `train` on a subset of data but get predictions for all of them.
Not sure whether this is actually possible / sensible though.

Reply:
this should already be possible - i added a toy example in the tests